src/components/TodoFooter/TodoFooter.tsx

```diff
@@ -0,0 +1,88 @@
+import { FC, memo, useMemo } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { FilterType } from '../../types/FilterType';
+
+type Props = {
+  todos: Todo[],
+  filterType: FilterType
+  handleFilterTypeChange: (filterType: FilterType) => void,
+  handleRemoveTodo: (id: number) => void,
+};
+
+export const TodoFooter: FC<Props> = memo((props) => {
+  const {
+    todos,
+    filterType,
+    handleFilterTypeChange,
+    handleRemoveTodo,
+  } = props;
+
+  const completedTodos = useMemo(() => (
+    todos.filter(todo => todo.completed)
+  ), [todos]);
+
+  const handleRemoveCompletedTodos = () => {
+    todos.forEach(todo => {
+      if (todo.completed) {
+        handleRemoveTodo(todo.id);
+      }
+    });
+  };
```

Comment:
let's wrap it with `useCallback`

Reply:
let's do it with `Promise.all` (i believe removing todo is async function)

<img width="622" alt="image" src="https://user-images.githubusercontent.com/55802630/187148770-76097c86-ddf5-4d5a-a67f-23bc0055f82d.png">

Reply:
> let's do it with `Promise.all` (i believe removing todo is async function)
> 
> <img alt="image" width="622" src="https://user-images.githubusercontent.com/55802630/187148770-76097c86-ddf5-4d5a-a67f-23bc0055f82d.png">

I had done with Promise.all, but then found that it is possible to handle with other methods😁 I Refactored one more time.
Also, I think it is unnecessary to wrap with useCallback if the function I use only in the same component where it is declared, but ok.